server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodServer.java

```diff
@@ -117,13 +126,20 @@ public HotRodServer() {
       super("HotRod");
    }
 
+   private static Map<String, CacheInfo> expirationMap(long time, TimeUnit unit) {
+      com.github.benmanes.caffeine.cache.Cache<String, CacheInfo> cache = Caffeine.newBuilder()
+            .expireAfterWrite(time, unit).build();
+      return cache.asMap();
+   }
+
    private Address clusterAddress;
    private ServerAddress address;
    private Cache<Address, ServerAddress> addressCache;
    private Map<String, AdvancedCache> knownCaches = CollectionFactory.makeConcurrentMap(4, 0.9f, 16);
    private Map<String, Configuration> knownCacheConfigurations = CollectionFactory.makeConcurrentMap(4, 0.9f, 16);
    private Map<String, ComponentRegistry> knownCacheRegistries = CollectionFactory.makeConcurrentMap(4, 0.9f, 16);
-   private final Map<String, CacheInfo> cacheInfo = new ConcurrentHashMap<>();
+   // This needs to be an expiration map so we can update knowledge about if sync listeners are registered
```

Comment:
Why do you need expiration exactly?

Reply:
So the reason for expiration is due to the fact that this `CacheInfo` is cached. Before this PR this information was static to the Cache configuration. However now that we require checking if a sync listener is installed, which can change over time, we need a way to recheck this every so often.

There may be a way to plug into listener installation to see if one is sync, but I have my doubts about that working.

And as I mentioned in my below comment, when I refactor listeners for 10.0 this is no longer needed and can be reverted to a regular `ConcurrentHashMap`